polygon/bridge/bridge.go

```diff
@@ -0,0 +1,200 @@
+package bridge
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ledgerwatch/log/v3"
+
+	libcommon "github.com/ledgerwatch/erigon-lib/common"
+	"github.com/ledgerwatch/erigon-lib/kv"
+	"github.com/ledgerwatch/erigon/common/u256"
+	"github.com/ledgerwatch/erigon/core"
+	"github.com/ledgerwatch/erigon/core/state"
+	"github.com/ledgerwatch/erigon/core/types"
+	"github.com/ledgerwatch/erigon/node"
+	"github.com/ledgerwatch/erigon/node/nodecfg"
+	"github.com/ledgerwatch/erigon/polygon/bor/borcfg"
+	"github.com/ledgerwatch/erigon/polygon/heimdall"
+)
+
+type fetchSyncEventsType func(ctx context.Context, fromId uint64, to time.Time, limit int) ([]*heimdall.EventRecordWithTime, error)
+
+type IDRange struct {
+	start, end uint64
+}
+
+type Bridge struct {
+	DB                       kv.RwDB
```

Comment:
best to keep the db as a private/unexported field - external components should not be able to access the DB directly but only via the Bridge API